support/builder/build_dep_populate.sh

```diff
@@ -0,0 +1,463 @@
+
+genBldrCreds() {
+    echo ""
+    echo "Please provide a Builder Auth Token."
+    read -r -p "Builder Auth token: " bldr_auth_token
+        export HAB_AUTH_TOKEN="${bldr_auth_token}"
+    echo ""
+    echo "Builder auth token set to: ${HAB_AUTH_TOKEN}"
+    read -r -p "Would you like to use this auth token? [y/N]" response
+    if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+        return
+    else
+        genBldrCreds
+    fi
+}
+
+setBldrUrl() {
+    echo ""
+    echo "Package ingestion can be pointed towards public builder or"
+    echo "an on-prem builder instance."
+
+    if [[ -n ${HAB_BLDR_URL} ]]; then
+       echo ""
+       echo "Builder URL configured via ENVVAR detected."
+       echo ""
+       echo "HAB_BLDR_URL=${HAB_BLDR_URL}"
+       export bldr_url="${HAB_BLDR_URL}"
+    else
+        echo ""
+        read -r -p "Will you be uploading to public builder? [y/N]" response
+            if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                export bldr_url="https://bldr.habitat.sh"
+            else
+                echo ""
+                echo "Please provide the URL so your builder instance"
+                echo "Ex: https://bldr.habitat.sh or http://localhost"
+                read -r -p ": " response
+                    export bldr_url="${response}"
+                echo ""
+                echo "Your builder URL is now configured to ${bldr_url}"
+                read -r -p "Is this correct? [y/N]" response
+                    if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                        return
+                    else
+                        setBldrUrl
+                    fi
+        fi
+    fi
+}
+
+credSelect() {
+    echo ""
+    read -r -p "Would you like to use these credentials? [y/N] " response
+    if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+        return
+    else
+        echo ""
+        read -r -p "Would you like to configure with custom credentials now? [y/N]" response
+        if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+            genS3Config
+        else
+            echo "Please reconfigure your AWS credentials and re-run s3migrate."
+            exit 1
+        fi
+    fi
+}
+
+credCheck(){
+    CREDS=( "${HOME}/.aws/credentials" "${HOME}/.aws/config" "/root/.aws/credentials" "/root/.aws/config")
+    for location in "${CREDS[@]}"; do
+        if [ -f "${location}" ]; then
+            echo ""
+            echo "AWS Credentials file located at ${location}"
+            cat "${location}"
+            credSelect
+            credsConfigured=true
+            break
+        fi
+
+        if [[ -n ${AWS_ACCESS_KEY_ID:-} ]]; then
+            if [[ -n ${AWS_SECRET_ACCESS_KEY:-} ]]; then
+                echo ""
+                echo "AWS Credentials configured via ENVVAR detected."
+                echo ""
+                echo "aws_access_key_id=${AWS_ACCESS_KEY_ID:-}"
+                echo "aws_secret_access_key=${AWS_SECRET_ACCESS_KEY:-}"
```

Comment:
We know based on the above `-n` test that `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` and non-empty strings, so the `:-` is superfluous and may hide errors later if the code is changed.